app/test/widgets/filter_button_test.dart

```diff
@@ -11,23 +11,37 @@ void main() {
     Modular.init(TestModule());
   });
 
-  testWidgets('Test FilterButton color change when markets are selected', (tester) async {
-    final filtersStore = Modular.get<FiltersStore>();
+  group('FilterButton widget', () {
+    testWidgets('FilterButton\'s color changes when market(s) are selected', (tester) async {
+      final filtersStore = Modular.get<FiltersStore>();
+      await tester.pumpWidget(
+        const MaterialApp(
+            home: Scaffold(
+          body: FilterButton(),
+        )),
+      );
 
-    await tester.pumpWidget(
-      const MaterialApp(
-          home: Scaffold(
-        body: FilterButton(),
-      )),
-    );
+      expect(find.byType(ElevatedButton), findsOneWidget);
+      // The following "resolve" method is used to get the color value based on the current state of the widget
+      expect(tester.widget<ElevatedButton>(find.byType(ElevatedButton)).style!.backgroundColor!.resolve({}), Colors.grey);
+      filtersStore.changeSelectedMarkets(Market(id: 0, name: 'Walmart', website: 'walmart.com'), true);
+      await tester.pump();
+      expect(tester.widget<ElevatedButton>(find.byType(ElevatedButton)).style!.backgroundColor!.resolve({}), Colors.blue);
+    });
 
-    expect(find.byType(ElevatedButton), findsOneWidget);
+    testWidgets('A modal is open when the user clicks on FilterButton', (tester) async {
```

Comment:
Looks like the tester function is duplicated here

Reply:
Ok

Reply:
Can you believe that during the second test, when I remove the duplicate the test starts failing?

Reply:
FIXED. Problem was the if statement that needed to check if marketStore.markets was not empty.